catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -314,17 +278,89 @@ protected Object executeWithSubject() throws Exception {
         return null;
     }
 
+    private File getInputFile() {
```

Comment:
Instead of returning `null`, did you consider changing this to return `Optional<File>`?